python/pyarrow/tests/test_array.py

```diff
@@ -3192,3 +3192,41 @@ def test_to_pandas_timezone():
     arr = pa.chunked_array([arr])
     s = arr.to_pandas()
     assert s.dt.tz is not None
+
+
+@pytest.mark.parametrize("arr", (
+    [None, [0]],
+    [None, [0, None], [0]],
+    [[0], [1]],
+)
+)
+def test_list_array_from_arrays(arr):
```

Comment:
There is already an existing `test_list_from_arrays` test, can you maybe move this one to just after the existing one? (to keep related tests close)